automatilib/cola_auth/i_dot_ai_cola_backend.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.backends import ModelBackend
+
+LOGGER = logging.getLogger(__name__)
+
+User = get_user_model()
+
+
+class AuthenticationBackend(ModelBackend):
+    def authenticate(self, request, username=None, password=None, **kwargs):
+        """
+        Get user response given by JWT token from kwargs and update the user based on this
+        :param request: The HTTP request that is requesting authentication
+        :param username: The users username (not used)
+        :param password: The users password (not used)
+        :param kwargs: The kwargs dict containing the JWT token body
+        :return: The user object
+        """
+        user_response = kwargs["user_response"]
+        user, created = User.objects.get_or_create(
+            email=user_response.items()["email"],
+        )
+        user.save()
+        LOGGER.info(f"Set values: {user_response} - from COLA authentication")
+        return user
+
+    def get_user(self, user_id):
+        try:
+            return User.objects.get(pk=user_id)
+        except User.DoesNotExist:
+            return None
```

Comment:
Would it be more accurate to return the DoesNotExist exception?

Reply:
We currently do this if a user is not found

```
LOGGER.error("No user found")
return HttpResponseServerError()
```

Reply:
do we need this override https://github.com/django/django/blob/fdd1323b9c83e56184e0c992af8faf8d54327775/django/contrib/auth/backends.py#L155?

Reply:
Sorry only just seen this comment. Apparently yes, though I'm completely unsure why, django must just really like this one method

https://reintech.io/blog/writing-custom-authentication-backend-django